csrc/runtime/allocations.cpp

```diff
@@ -670,7 +670,12 @@ TensorShapeInfo inferTensorShapes(
     const ExpressionEvaluator& expr_eval) {
   // Alias handling:
   auto alias_info = tv->fusion()->getOutputAlias(tv);
-  if (alias_info.type == AllocationType::Evaluate) {
+  if (alias_info.type != AllocationType::New) {
+    // For reuse buffer alias, we need to get the aliased_io's size/stride
+    if (alias_info.type == AllocationType::ReuseBuffer) {
+      tv = alias_info.aliased_io->as<TensorView>();
```

Comment:
Without this patch, we'll ended up with passing the wrong kernel arguments when we have inplace updates on inputs with a non-trivial allocation domain.

The earlier refactor PR broke this test: https://github.com/NVIDIA/Fuser/blob/5111d3b4db305937b5eb0810c1193efac6c3aa37/tests/python/test_python_frontend.py#L4949-L4950

Unfortunately the two PRs ( the refactor #3952 and the test #3970 ) were merged in parallel and never combined in the CI.